### ISSUE TITLE
mingw: new build warnings in dshow module

### ISSUE THREAD

**Issue by alfredh** [baresip/baresip#2304]
Build warnings with mingw and dshow:

```
2022-11-03T10:41:32.9804256Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp: In member function ‘virtual HRESULT Grabber::BufferCB(double, BYTE*, long int)’:
2022-11-03T10:41:32.9805781Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:142:59: warning: unused parameter ‘buf_len’ [-Wunused-parameter]
2022-11-03T10:41:32.9806497Z   142 |  STDMETHOD(BufferCB) (double sample_time, BYTE *buf, long buf_len)
2022-11-03T10:41:32.9807064Z       |                                                      ~~~~~^~~~~~~
2022-11-03T10:41:32.9808597Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp: In member function ‘virtual HRESULT Grabber::SampleCB(double, IMediaSample*)’:
2022-11-03T10:41:32.9809831Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:169:30: warning: unused parameter ‘sample_time’ [-Wunused-parameter]
2022-11-03T10:41:32.9810399Z   169 |  STDMETHOD(SampleCB) (double sample_time, IMediaSample *samp)
2022-11-03T10:41:32.9810775Z       |                       ~~~~~~~^~~~~~~~~~~
2022-11-03T10:41:32.9811427Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:169:57: warning: unused parameter ‘samp’ [-Wunused-parameter]
2022-11-03T10:41:32.9811955Z   169 |  STDMETHOD(SampleCB) (double sample_time, IMediaSample *samp)
2022-11-03T10:41:32.9812348Z       |                                           ~~~~~~~~~~~~~~^~~~
2022-11-03T10:41:32.9847254Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp: In function ‘void destructor(void*)’:
2022-11-03T10:41:32.9849086Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:475:13: warning: deleting object of polymorphic class type ‘Grabber’ which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
2022-11-03T10:41:32.9852175Z   475 |  delete st->grab;
2022-11-03T10:41:32.9852786Z       |             ^~~~
2022-11-03T10:41:32.9866153Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp: In function ‘int alloc(vidsrc_st**, const vidsrc*, vidsrc_prm*, const vidsz*, const char*, const char*, void (*)(vidframe*, uint64_t, void*), void (*)(vidpacket*, void*), void (*)(int, void*), void*)’:
2022-11-03T10:41:32.9867508Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:482:16: warning: unused parameter ‘fmt’ [-Wunused-parameter]
2022-11-03T10:41:32.9868006Z   482 |    const char *fmt, const char *dev,
2022-11-03T10:41:32.9868362Z       |    ~~~~~~~~~~~~^~~
2022-11-03T10:41:32.9869372Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:484:22: warning: unused parameter ‘packeth’ [-Wunused-parameter]
2022-11-03T10:41:32.9870031Z   484 |    vidsrc_packet_h  *packeth,
2022-11-03T10:41:32.9870424Z       |    ~~~~~~~~~~~~~~~~~~^~~~~~~
```

**Comment by alfredh** [comment]
Current warnings:

```
2022-11-03T11:39:29.6676773Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:17: warning: ignoring #pragma comment  [-Wunknown-pragmas]
2022-11-03T11:39:29.6677420Z    17 | #pragma comment(lib, "strmiids.lib")
2022-11-03T11:39:29.6677762Z       | 
2022-11-03T11:39:30.4336029Z make[3]: Leaving directory '/home/runner/work/baresip/baresip/baresip-win32/baresip/build'
2022-11-03T11:39:30.4490766Z [ 45%] Built target wincons
2022-11-03T11:39:30.7259767Z make[3]: Leaving directory '/home/runner/work/baresip/baresip/baresip-win32/baresip/build'
2022-11-03T11:39:30.7272190Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp: In function ‘void destructor(void*)’:
2022-11-03T11:39:30.7273860Z /home/runner/work/baresip/baresip/baresip-win32/baresip/modules/dshow/dshow.cpp:478:13: warning: deleting object of polymorphic class type ‘Grabber’ which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
2022-11-03T11:39:30.7274992Z   478 |  delete st->grab;
2022-11-03T11:39:30.7275637Z       |             ^
~~~


Any idea how to fix the pragma warning ?

**Comment by alfredh** [comment]
fixed in 964a19ac64baa96b6d6e97fed8606d9deb7fc1a4
